packages/xo-server/src/api/vm.js

```diff
@@ -1670,3 +1670,24 @@ deleteVgpu.params = {
 deleteVgpu.resolve = {
   vgpu: ['vgpu', 'vgpu', ''],
 }
+
+// -------------------------------------------------------------------
+
+export async function blockDeletion({ vm, block = true }) {
```

Comment:
This method only modify a VM field, it don't do more actions than that. So, it can be included  in `VM.set`.